app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt

```diff
@@ -82,6 +92,30 @@ class PreviewBadge : View {
                 View.MeasureSpec.makeMeasureSpec(calculatedHeight, View.MeasureSpec.EXACTLY))
     }
 
+    public override fun onSaveInstanceState(): Parcelable? {
+        val bundle = Bundle()
+        bundle.putParcelable(BUNDLE_STATE, super.onSaveInstanceState())
+        bundle.putBoolean(BUNDLE_FLASH, this.ifFlash)
+        bundle.putBoolean(BUNDLE_MARQUEE, this.ifMarquee)
+        bundle.putInt(BUNDLE_SPEED, this.badgeSpeed)
+        bundle.putInt(BUNDLE_MODE, this.badgeMode.ordinal)
+        bundle.putParcelableArrayList(BUNDLE_CHECKLIST, ArrayList(this.checkList))
```

Comment:
```suggestion
        bundle.putParcelableArrayList(BUNDLE_CHECKLIST, this.checkList)
```

Reply:
I tried that sir, did not work. It only takes arraylist as an argument

Reply:
and there was no function for lists or mutable lists

Reply:
Then change checkList to ArrayList

Reply:
Alright Sir

Reply:
Done @iamareebjamal